dbus/methods_test.go

```diff
@@ -130,6 +133,238 @@ func TestStartStopUnit(t *testing.T) {
 	}
 }
 
+// Ensure that ListUnitsByNames works.
+func TestListUnitsByNames(t *testing.T) {
+	target1 := "systemd-journald.service"
+	target2 := "unexisting.service"
+
+	conn := setupConn(t)
+
+	units, err := conn.ListUnitsByNames([]string{target1, target2})
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var unit *UnitStatus
+	for _, u := range units {
+		if u.Name == target1 {
+			unit = &u
+			break
+		}
+	}
+
+	if unit == nil {
+		t.Fatalf("%s unit not found in list", target1)
+	}
+
+	if unit.ActiveState != "active" {
+		t.Fatalf("%s unit should be active but it is %s", target1, unit.ActiveState)
+	}
+
+	unit = nil
+	for _, u := range units {
+		if u.Name == target2 {
+			unit = &u
+			break
+		}
+	}
+
+	if unit == nil {
```

Comment:
Oh... confusing

On Tue, May 31, 2016 at 5:34 PM, kayrus <notifications@github.com> wrote:

> In dbus/methods_test.go
> <https://github.com/coreos/go-systemd/pull/150#discussion_r65206518>:
>
> > +		t.Fatalf("%s unit not found in list", target1)
> > +	}
> > +
> > +	if unit.ActiveState != "active" {
> > +		t.Fatalf("%s unit should be active but it is %s", target1, unit.ActiveState)
> > +	}
> > +
> > +	unit = nil
> > +	for _, u := range units {
> > +		if u.Name == target2 {
> > +			unit = &u
> > +			break
> > +		}
> > +	}
> > +
> > +	if unit == nil {
>
> @jonboulle <https://github.com/jonboulle> take a look on this comment
> https://github.com/coreos/go-systemd/pull/150/files#diff-b5b8384be79564335a3b50bcfed63e91R293
> systemd should return unit states for non-existing units.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/coreos/go-systemd/pull/150/files/124d391a9338d39c19cd55de93b70a4e7ed649f4#r65206518>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACewNzMPezWMqZoW3zMavEVcKRxk5tcbks5qHFUSgaJpZM4IMyF5>
> .
>